cmd/thanos/receive.go

```diff
@@ -258,6 +259,15 @@ func runReceive(
 	// hashringChangedChan signals when TSDB needs to be flushed and updated due to hashring config change.
 	hashringChangedChan := make(chan struct{}, 1)
 
+	level.Debug(logger).Log("msg", "setting up hashring")
+	{
+		if hashringConfigs, err := setupHashring(g, logger, reg, conf, hashringChangedChan, webHandler, statusProber, enableIngestion); err != nil {
+			return err
+		}
+		
+		dbs.hashringConfigs = hashringConfigs
```

Comment:
This is an unexported field, so you won't be able to assign it here.
```suggestion
		dbs.HashringConfigs = hashringConfigs
```

Reply:
Also, why do it like this when we have NewMultiTSDB function take input?

Reply:
In the first approach, `NewMultiTSDB` is called before `HashringConfigs` are loaded. But again, we agreed to reject this approach...

Reply:
Yup, was just pointing out the error in case you were trying to get it to work. 🙂

Reply:
Sounds time-consuming and nonsense. But I'll try to fix all the errors in the first approach (very close now), check locally to see it fail, only then move to the second approach. I want to actually see it fail. Hopefully will learn sth.